docs/csharp/quick-starts/branches-and-loops.yml

```diff
@@ -0,0 +1,241 @@
+    use loops to repeat statements that you want executed more than once. Try
+    this code in the interactive window:
+
+    ```csharp
+    int counter = 0;
+    while (counter < 10)
+    {
+      Console.WriteLine("Hello World! The counter is " + counter);
+      counter++;
+    }
+    ```
+
+    The `while` statement checks a condition and executes the statement
+    following the `while`. It will repeat checking the condition and
+    executing those statements until the condition is false.
+
+    There's one other new operator in this example. The `++` after
+    the `counter` variable is the **increment** operator. It adds 1
+    to the value of counter, and stores that value in the counter variable.
+
+    > [!IMPORTANT]
+    > Make sure that the `while` loop condition does switch to
+    > false as you execute the code. Otherwise, you create an
+    > **infinite loop** where your program never ends. Let's
+    > not demonstrate that, because the engine that runs your
+    > code will timeout and you'll see no output from your program.
+
+    The `while` loop tests the condition before executing the code
+    following the `while`. The `do` ... `while` loop executes the 
+    code first, and then checks the condition. It looks like this:
+
+    ```csharp
+    int counter = 0;
+    do
+    {
+      Console.WriteLine("Hello World! The counter is " + counter);
+      counter++;
+    } while (counter < 10);
+    ```
+    
+    This `do` loop and the earlier `while` loop work the same. 
+    
+    Let's move on to one last loop statement.
+
+- title: Work with the `for` loop.
+  durationInMinutes: 4
+  content: |
+    Another common loop statement that you'll see in C# code is the
+    `for` loop. Try this code in the interactive window:
+
+    ```csharp
+    for(int counter = 0; counter < 10; counter++)
+    {
+      Console.WriteLine("Hello World! The counter is " + counter);
+    } 
+    ```
+
+    This does the same work as the `while` loop and the `do` loop you've
+    already used. The `for` statement has three parts that control
+    how it works. 
+
+    The first part is the **for initializer**: `for counter = 0;` declares
+    that `counter` is the loop variable, and sets its initial value to `0`.
+
+    The middle part is the **for condition**: `counter < 10` declares that this
+    `for` loop continues to execute as long as the value of counter is less than 10.
+
+    The final part is the **for iterator**: `counter++` specifies how to modify the loop
+    variable after executing the block following the `for` statement. Here, it specirifies
+    that `counter` should be incremented by 1 each time the block executes.
+
+    Experiment with these yourself. Try each of the following:
+
+    - Change the initializer to start at a different value.
+    - Change the condition to stop at a different value.
+    - Change the iterator to add a value other than 1 to `counter`.
```

Comment:
isn't it too early to ask this? most people will use counter++ or counter-- in their for loops. they have just been introduced to counter++. will they know they should self assign a value here like counter += 2?

Reply:
Good point.